File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/RelationV1.java
package com.cleverferret.v2.core.database.schema;

import com.cleverferret.v2.core.common.VersionedContract;

/**
 * Canonical edge row joining media with collections/tags/series/other media.
 */
public record RelationV1(
        String relationId,
        RelationTypeV1 relationType,
        String fromEntityType,
        String fromEntityId,
        String toEntityType,
        String toEntityId,
        Integer ordinal,
        String role,
        String sourceSystem,
        long createdAtEpochMs,
        long updatedAtEpochMs
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/RelationV1.kt
package com.cleverferret.v2.core.database.schema

import com.cleverferret.v2.core.common.VersionedContract

data class RelationV1(
    val relationId: String,
    val relationType: RelationTypeV1,
    val fromEntityType: String,
    val fromEntityId: String,
    val toEntityType: String,
    val toEntityId: String,
    val ordinal: Int?,
    val role: String,
    val sourceSystem: String,
    val createdAtEpochMs: Long,
    val updatedAtEpochMs: Long
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
